buildSrc/src/main/kotlin/TargetsConfig.kt

```diff
@@ -89,205 +106,13 @@ fun Project.configureTargets() {
-                val tvosMain by getting
-                val tvosTest by getting
-
-                nixMain?.let { darwinMain.dependsOn(it) }
-                macosMain.dependsOn(darwinMain)
-                tvosMain.dependsOn(darwinMain)
-                iosMain.dependsOn(darwinMain)
-                watchosMain.dependsOn(darwinMain)
-
-                macosTargets().forEach {
-                    getByName("${it}Main").dependsOn(macosMain)
-                    getByName("${it}Test").dependsOn(macosTest)
-                }
-
-                iosTargets().forEach {
-                    getByName("${it}Main").dependsOn(iosMain)
-                    getByName("${it}Test").dependsOn(iosTest)
-                }
-
-                watchosTargets().forEach {
-                    getByName("${it}Main").dependsOn(watchosMain)
-                    getByName("${it}Test").dependsOn(watchosTest)
-                }
-
-                tvosTargets().forEach {
-                    getByName("${it}Main").dependsOn(tvosMain)
-                    getByName("${it}Test").dependsOn(tvosTest)
-                }
-
-                darwinTargets().forEach {
-                    getByName("${it}Main").dependsOn(darwinMain)
-                    getByName("${it}Test").dependsOn(darwinTest)
-                }
-            }
-
-            if (hasLinux) {
-                val linuxMain by getting {
-                    findByName("nixMain")?.let { dependsOn(it) }
-                }
-
-                val linuxTest by getting {
-                    findByName("nixTest")?.let { dependsOn(it) }
-
-                    dependencies {
-                        implementation(kotlin("test"))
-                    }
-                }
-
-                linuxTargets().forEach {
-                    getByName("${it}Main").dependsOn(linuxMain)
-                    getByName("${it}Test").dependsOn(linuxTest)
-                }
-            }
-
-            if (hasDesktop) {
-                val desktopMain by getting {
-                    findByName("posixMain")?.let { dependsOn(it) }
-                }
-
-                val desktopTest by getting
-
-                desktopTargets().forEach {
-                    getByName("${it}Main").dependsOn(desktopMain)
-                    getByName("${it}Test").dependsOn(desktopTest)
-                }
-            }
-
-            if (hasWindows) {
-                val windowsMain by getting {
-                    findByName("posixMain")?.let { dependsOn(it) }
-                }
-
-                val windowsTest by getting {
-                    dependencies {
-                        implementation(kotlin("test"))
-                    }
-                }
-
-                windowsTargets().forEach {
-                    getByName("${it}Main").dependsOn(windowsMain)
-                    getByName("${it}Test").dependsOn(windowsTest)
-                }
-            }
+            if (hasJvmAndNix) { jvm(); nixTargets(); }
```

Comment:
just minor unrelated remark: those functions (nixTargets/posixTargets/etc) are extensions on `Project` - so it would be more clear to call them at least at `kotlin` extension level (because of `jvm` function call).
Even better to refactor them to be extensions on `KotlinMultiplatformExtension` - but of course it's out of scope here :)

Reply:
agree